virttest/migration.py

```diff
@@ -0,0 +1,326 @@
+                     'func_param' is mandatory if no real func_param, none is
+                     requested.
+                     'shell' is optional, where shell=True(bool) can be used
+                     for process.run
+
+        """
+        if migration_type == "orderly":
+            for vm in vms:
+                migration_thread = threading.Thread(target=self.thread_func_migration,
+                                                    args=(vm, desturi, options,
+                                                          ignore_status, virsh_opt,
+                                                          extra_opts))
+                migration_thread.start()
+                eclipse_time = 0
+                stime = int(time.time())
+                if func:
+                    # Execute command once the migration is started
+                    migrate_start_state = args.get("migrate_start_state", "paused")
+                    if self.wait_for_migration_start(vm, state=migrate_start_state, uri=desturi):
+                        logging.info("Migration started for %s", vm.name)
+                        if func == process.run:
+                            try:
+                                func(args['func_params'], shell=args['shell'])
+                            except KeyError:
+                                func(args['func_params'])
+                        elif func == virsh.migrate_postcopy:
+                            func(vm.name, uri=srcuri, debug=True)
+                        else:
+                            func(args['func_params'])
+                    else:
+                        logging.error("Migration failed to start for %s",
+                                      vm.name)
+                eclipse_time = int(time.time()) - stime
+                logging.debug("start_time:%d, eclipse_time:%d", stime, eclipse_time)
+                if eclipse_time < thread_timeout:
+                    migration_thread.join(thread_timeout - eclipse_time)
+                if migration_thread.isAlive():
+                    logging.error("Migrate %s timeout.", migration_thread)
+                    self.RET_LOCK.acquire()
+                    self.RET_MIGRATION = False
+                    self.RET_LOCK.release()
+        elif migration_type == "cross":
+            # Migrate a vm to remote first,
+            # then migrate another to remote with the first vm back
+            vm_remote = vms.pop()
+            self.thread_func_migration(vm_remote, desturi)
+            for vm in vms:
+                thread1 = threading.Thread(target=self.thread_func_migration,
+                                           args=(vm_remote, srcuri, options))
+                thread2 = threading.Thread(target=self.thread_func_migration,
+                                           args=(vm, desturi, options))
+                thread1.start()
+                thread2.start()
+                thread1.join(thread_timeout)
```

Comment:
I hope this `thread_timeout` should be same or sightly more than migration timeout, so it would be good to use migration timeout as a variable to define thread_timeout, consider  migration takes a longer time and that is being handled through `options`, and leaving `thread_timeout` as default 60 secs won't suffice?